code/game/turfs/simulated.dm

```diff
@@ -11,23 +15,53 @@
 	var/dirt = 0
 
 // possible fix for the perma wet floor bug
- /turf/simulated/proc/wet_floor(var/wet_val = 1)
- 	spawn(0)
- 		if(wet_val <= wet)
- 			return
- 		wet = wet_val
- 		if(wet_overlay)
- 			overlays -= wet_overlay
- 			wet_overlay = null
- 		wet_overlay = image('icons/effects/water.dmi',src,"wet_floor")
- 		overlays += wet_overlay
- 		sleep(800)
- 		if(wet >= 2)
- 			sleep(1600)
- 		wet = 0
- 		if(wet_overlay)
- 			overlays -= wet_overlay
- 			wet_overlay = null
+/turf/simulated/wet_floor(var/wet_val = 1, var/duration)
+	if (wet_val > wet)
+		switch(wet_val)
+			if (1)
+				slip_dist_power = 75
+				slip_stun = 4
+				wet_descriptor = "wet"
+			if (2)
+				slip_dist_power = 92
+				slip_stun = 8
+				wet_descriptor = "slippery"
+			if (3)
+				slip_dist_power = 85
+				slip_stun = 6
+				wet_descriptor = "icy"
+			else
+				return //some invalid value was passed
+		wet = wet_val
+
+	wetness_duration += duration
+
+	if (wet_overlay)
+		overlays -= wet_overlay
+		wet_overlay = null
+	processing_objects |= src

```

Comment:
Boolean `is_processing` and
```
if (!is_processing)
    processing_objects += src
    is_processing = 1
```

is faster. Primarily because the `|=` operator looks through the list first, to find duplicates. On average, the size of the list it looks through is around 600 - 750 entries long.